azure/scope/cluster.go

```diff
@@ -230,18 +231,22 @@ func (s *ClusterScope) RouteTableSpecs() []azure.RouteTableSpec {
 }
 
 // NatGatewaySpecs returns the node nat gateway.
-func (s *ClusterScope) NatGatewaySpecs() []azure.NatGatewaySpec {
-	natGateways := []azure.NatGatewaySpec{}
+func (s *ClusterScope) NatGatewaySpecs() []azure.ResourceSpecGetter {
+	natGateways := []azure.ResourceSpecGetter{}
 
 	// We ignore the control plane nat gateway, as we will always use a LB to enable egress on the control plane.
+	// TODO: do we need to use setSubnet() in natgateways.go, since it doesn't add/remove any NodeSubnets() or change IsNatGatewayEnabled()
 	for _, subnet := range s.NodeSubnets() {
```

Comment:
if we switch to updating the ID for every subnet that has this NAT GW in its spec, we might also be able to de-dupicate the NAT gateways, and only add it to spec if it wasn't already added.